File: Codellama-34b-Instruct/Template_and_context_final/serial_construct_reduction_clause.f90
fortran
#ifndef T1
!T1:serial construct reduction clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: i, j, k
        REAL :: a(10), b(10), c(10)
        REAL :: sum
        
        !$acc serial
        !$acc reduction(+:sum)
        DO i = 1, 10
          a(i) = i
          b(i) = i * 2
          c(i) = i * 3
        END DO
        !$acc end serial
        
        sum = 0
        DO i = 1, 10
          sum = sum + a(i) + b(i) + c(i)
        END DO
        
        IF (sum .ne. 120) THEN
          errors = errors + 1
        END IF
        
        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: Phind-CodeLlama-34B-v2/context-template-full/serial_construct_reduction_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
/*T1:serial construct reduction clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);

    int array[100];
    int host_result = 0;
    for (int x = 0; x < 100; ++x) {
        array[x] = rand() % 100;
        host_result += array[x];
    }

    #pragma acc serial reduction(+:host_result)
    {
        for (int x = 0; x < 100; ++x) {
            host_result += array[x];
        }
    }

    int device_result = 0;
    #pragma acc parallel copy(array[0:100]) reduction(+:device_result)
    {
        for (int x = 0; x < 100; ++x) {
            device_result += array[x];
        }
    }

    if (host_result != device_result) {
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}